tests/network/primary_pod_network.go

```diff
@@ -0,0 +1,137 @@
+/*
+ * This file is part of the KubeVirt project
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * Copyright 2020 Red Hat, Inc.
+ *
+ */
+
+package network
+
+import (
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	v1 "kubevirt.io/client-go/api/v1"
+	"kubevirt.io/client-go/kubecli"
+
+	"kubevirt.io/kubevirt/tests"
+	cd "kubevirt.io/kubevirt/tests/containerdisk"
+)
+
+var _ = SIGDescribe("Primary Pod Network", func() {
+	var virtClient kubecli.KubevirtClient
+
+	BeforeEach(func() {
+		var err error
+		virtClient, err = kubecli.GetKubevirtClient()
+		tests.PanicOnError(err)
```

Comment:
Is there a specific reason to use this "Panic" instead of a regular assert?

Reply:
I think it is the only way to crash if we are in a goroutine. It is not needed here, changing to Expect.

Reply:
Hmm.. I was under the impression that the assertions are panics, otherwise how can they work?

Reply:
You are right, they are. I did not explained it clearly. The thing is that for Expect to work correctly and be recognized by Ginkgo, you need to register panic recover per each spawned goroutine using `defer GinkgoRecover()` [1]. If you call Expect without the recover being set, it may give you unexpected output. With panic, you make sure that it clearly explodes without Expect formatting being involved.

Anyhow, this is my guess for the motivation to use PanicOnError in tests as a universal "exit" for unrecoverable invalid states in any goroutine. It does not make sense to use it in this case, so I dropped it.

[1] https://onsi.github.io/ginkgo/

Reply:
Thank you for the detailed explanation. I would prefer not to see it ever, even in goroutines. I hope to see asserts only in the tests or fixtures level, not in helpers, making this a non-issue.